File: 0328-odd-even-linked-list/0328-odd-even-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var oddEvenList = function (head) {
    if (head==null || head.next==null || (head.next.next==null)){
        return head
    }
    var even = head;
    var odd = head.next;
    var temp = head.next;
    while (odd  && odd.next) {
        even.next =odd.next;
        even = odd.next;
        odd.next = even.next;
        odd = even.next;
        even.next=null;
    }
    even.next = temp;
    console.log(head.next)
    

    return head


};

File: 0328-odd-even-linked-list/0328-odd-even-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* oddEvenList(ListNode* head) {
        ListNode* current = head;
        ListNode* dummynodeodd = new ListNode(-1);
        ListNode* currentodd = dummynodeodd;
        ListNode* dummynodeeven = new ListNode(-1);
        ListNode* currenteven = dummynodeeven;
        int index=0;
        while(current!= NULL){
            if(index % 2==0){
                currentodd->next = current;
                currentodd = currentodd->next;
            }
            else{
                currenteven->next = current;
                currenteven= currenteven->next;

            }
            current = current->next;
            index++;
        }
        currentodd->next = dummynodeeven->next;
        currenteven->next = NULL;
        return dummynodeodd->next;

    }
};